docs/Users/Tutorials/External_Bears.rst

```diff
@@ -0,0 +1,521 @@
+
+
+    @external_bear_wrap(executable='coalaCheck_cpp',
+                        settings={})
+    class coalaCheckBear:
+        """
+        Checks for coala written with uppercase 'C'
+        """
+        LANGUAGES = {'All'}
+        REQUIREMENTS = {'coalaCheck_cpp'}
+        AUTHORS = {'Me'}
+        AUTHORS_EMAILS = {'me@mail.com'}
+        LICENSE = 'GPLv3'
+
+        @staticmethod
+        def create_arguments():
+            return ()
+
+6. Since the input will be a JSON string some kind of JSON class is needed.
+   `This one <https://github.com/nlohmann/json>`__ is a great choice because it
+   is easy to integrate and is used in this tutorial.
+7. Create ``coalaCheck.cpp`` and start by testing the input. The best thing
+   about nlohmann's JSON library is that you can parse JSON directly
+   from stdin like this:
+
+::
+
+    #include <iostream>
+
+    #include "json.hpp"
+
+    using json = nlohmann::json;
+    using namespace std;
+
+    json in;
+
+    int main() {
+
+        cin >> in;
+
+        cout << in;
+
+    return 0;
+
+8. Create a ``Makefile``. The JSON library requires C++11 so a sample
+   ``Makefile`` would look like this:
+
+::
+
+    build: coalaCheck.cpp
+        g++ -std=c++11 -o coalaCheck_cpp coalaCheck.cpp
+
+9. Compile and test the binary by giving it a JSON string. It should print the
+   JSON string back at ``stdout``.
+
+10. Read about the JSON Spec that the input uses(`The JSON Spec`_). The filename
+    is found in ``in["filename"]`` and the list of lines is found in
+    ``in["file"]``.
+11. Create a result adding function, also an init function proves quite useful
+    for initializing the output json.
+
+::
+
+    #include <iostream>
+    #include <string>
+
+    #include "json.hpp"
+
+    using json = nlohmann::json;
+    using namespace std;
+
+    json in;
+    json out;
+    string origin;
+
+    void init_results(string bear_name) {
+        origin = bear_name;
+        out["results"] = json::array({});
+    }
+
+    void add_result(string message, int line, int column, int severity) {
+        json result = {
+            {"origin", origin},
+            {"message", message},
+                {"affected_code", json::array({{
+                    {"file", in["filename"]},
+                    {"start", {
+                        {"column", column},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }},
+                    {"end", {
+                        {"column", column+6},
+                        {"file", in["filename"]},
+                        {"line", line}
+                    }}
+                }})},
+            {"severity", severity}
+        };
+        out["results"] += result;
+    }
+
+    int main() {
+
+        cin >> in;
+
+        init_results("coalaCheckBear");
+
+        cout << out;
+        return 0;
+    }
+
+The ``C++`` operators and syntax are not well suited for JSON manipulation but
```

Comment:
This is out of indentation. Either make it a Note or make it part of the above point and code with correct indentation.